src/overflow/depression_fill.py

```diff
@@ -0,0 +1,255 @@
+from heapq import heappush, heappop
+from enum import Enum
+import numpy as np
+from numba import njit
+from osgeo import gdal
+from overflow.fill_pits import (
+    priority_flood_tile,
+    handle_edge,
+    handle_corner,
+    make_sides,
+)
+from overflow.setup_bands import setup_bands
+from overflow.util.raster import RasterChunk, raster_chunker, read_tile
+
+
+class Coordinate:
+    row: np.int32
+    col: np.int32
+
+    def __init__(self, row, col):
+        self.row = row
+        self.col = col
+
+
+class Side(Enum):
+    TOP = 1
+    RIGHT = 2
+    BOTTOM = 3
+    LEFT = 4
+
+
+class Corner(Enum):
+    TOP_LEFT = 1
+    TOP_RIGHT = 2
+    BOTTOM_RIGHT = 3
+    BOTTOM_LEFT = 4
+
+
+class DemLabelArr:
+    labels: np.ndarray
+    dem: np.ndarray
+
+
+class DemLabel:
+    label: int
+    dem: float
+
+
+@njit
```

Comment:
I think we can't compile this main function, just the supporting functions, since for example io with gdal cannot be compiled with numba. We also can't call a non njit function from a njit function